schema_based_on_diagram.sql

```diff
@@ -0,0 +1,55 @@
+CREATE TABLE patients (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    name VARCHAR(255),
+    date_of_birth DATE,
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE medical_histories(
+    id INT GENERATED ALWAYS AS IDENTITY,
+    admitted_at TIMESTAMP,
+    patient_id INT REFERENCES patients(id),
+    status VARCHAR,
+    PRIMARY KEY(id) 
+);
+
+CREATE TABLE treatments(
+    id INT GENERATED ALWAYS AS IDENTITY,
+    type VARCHAR,
+    name VARCHAR,
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE invoices(
+    id INT GENERATED ALWAYS AS IDENTITY,
+    total_amount DECIMAL,
+    generated_at TIMESTAMP,
+    medical_history_id INT REFERENCES medical_histories(id),
+    PRIMARY KEY(id)
+);
```

Comment:
- [ ] It seems like you forgot to add the `payed_at` column which should hold a `timestamp` value. so please add it to align with the given diagram.